content/developer/howtos/upgrade_custom_db.rst

```diff
@@ -0,0 +1,305 @@
+
+[WIP] Empty database
+====================
+
+Before working on an upgraded test database, we recommend to make the custom developments work on an
+empty database in the targeted version of your upgrade.
+
+This ensures that the customization is compatible with the new version of Odoo, allows to analyse
+how does it behave and interact with the new features, and guarantees that they will not cause
+any issue when upgrading the database.
+Making the custom modules work in an empty database also helps avoiding changes and wrong
+configurations that might be present on the production database (like studio customization,
+customized website pages, mail templates or translations), which are not intrinsically related to the
+custom modules and that can raises unwanted issues in this stage of the upgraded process.
+
+To make custom modules work on an empty database we advise to follow this steps:
+
+  - :ref:`Make them installable <upgrade/empty_database/modules_installable>`
+  - :ref:`Test and fixes <upgrade/empty_database/test_fixes>`
+  - :ref:`Clean the code <upgrade/empty_database/clean_code>`
+  - :ref:`Make standard tests run successfully <upgrade/empty_database/standard_test>`
+
+.. TODO Check if needed
+.. At this stage, all the custom modules will be properly upgraded and fully functioning on an empty
+.. database and their code clean.
+.. It is ready to make the custom modules installable on the upgraded database.
+
+.. _upgrade/empty_database/modules_installable:
+
+Make custom modules installable
+-------------------------------
+
+The first step is to make the custom modules installable in the new Odoo version.
+This means, in a first instance, making sure there is no traceback or warnings when installing them.
+For this, install the custom modules (ideally, one by one) in an empty database targeting the new
+Odoo version and fix the tracebacks and warnings that arise from that.
+
+.. TODO Re-check and explain better the examples 
+
+This process will help detect issues during the installation of the modules.
+For example, invalid module dependencies, syntax changes (like the assets declaration, OWL updates,
+mail templates from Jinja to Qweb), non existing references to standard fields, models, views, etc.
+
+.. _upgrade/empty_database/test_fixes:
+
+Test and fixes
+--------------
+
+Once there are no more tracebacks when installing the modules, the next step is to test them.
+Even if the custom modules are installable in an empty database, this does not warranties there are
+no errors during their execution.
+Because of this, we encourage to test thoroughly all the customization to make sure everything
+is working as expected.
+
+This process will help detect further issues that are not identified during the module installation
+and can only be detected in runtime.
+For example, deprecated calls to standard python or OWL functions, non existing references to
+standard fields, etc.
+
+.. TODO Expand in the tests and their explanation
```

Comment:
IMO, we don't really need to. Any developer knows what testing is, I think that specifying what they should look at is enough.

Reply:
I'm not sure why I added this :thinking: 
That means we can remove it